components/konflux-ui/staging/base/proxy/nginx.conf

```
@@ -161,6 +161,25 @@ http {
             include /mnt/nginx-generated-config/bearer.conf;
         }
 
+        location ~* /api/k8s/api/v1/namespaces(/?)$ {
```

I think this can be simplified by dropping the regex expression. wdyt?

---

IIUC, If we remove this regex this will become the path every namespaced request will take, e.g. `api/v1/namespaces/my-ns/pods` will be processed by this rule

---

Correct.  We only want to match on requests for namespace resources, i.e. when the path is `/api/k8s/api/v1/namespaces` or `/api/k8s/api/v1/namespaces/`.

If I'm understanding [nginx's docs](https://nginx.org/en/docs/http/ngx_http_core_module.html#location) correctly, we could set it up like this and have it work:
```suggestion
        location = /api/k8s/api/v1/namespaces/ {
```

This would match the path we want, both with and without the trailing slash.  However, this has slightly different behavior than what we currently have: nginx will issue a 301 permanent redirect from `/api/k8s/api/v1/namespaces` to `/api/k8s/api/v1/namespaces/`, which I'm not sure I'm a fan of.  To me, the regex is clearer in its intent.

Also, if we _do_ decide to use regex matching here, we should probably do case-sensitive matching:
```suggestion
        location ~ /api/k8s/api/v1/namespaces(/?)$ {
```